pxd_fastpath.c

```diff
@@ -151,7 +151,7 @@ static int fastpath_global_threadctx_init(struct thread_context *tc, int cpuid)
 	for(;i >= 0; i--) {
 		if (tc->writer[i]) kthread_stop(tc->writer[i]);
 	}
-	i = PXD_MAX_THREAD_PER_CPU;
+	i = PXD_MAX_THREAD_PER_CPU - 1;
```

Comment:
Why not for fail_wr?

Reply:
because read threads are initialized first and then write threads. If there is a write thread init failure, then cleanup starts from the current index and goes back to zeroth index, then starts again to clean reads threads from the beginning. 
if read threads fail, then cleanup happens from that index and goes down to zero.
So the reinit to last index is needed only while cleaning partial write threads and all reads threads.